lib/active_record_extended/cte_deprecation_configuration.rb

```diff
@@ -0,0 +1,42 @@
+# frozen_string_literal: true
+
+module ActiveRecordExtended
+  class CteDeprecationConfiguration
```

Comment:
I would rather change this into a module `Configuration`. Not a fan of this explicit deprecation module since it's too limiting.